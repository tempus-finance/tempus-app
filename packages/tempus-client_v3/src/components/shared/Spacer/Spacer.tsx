import { FC } from 'react';

interface SpacerProps {
  size: number;
}

const Spacer: FC<SpacerProps> = props => {
  return <div style={{ width: `${props.size}px`, height: `${props.size}px` }} />;
};

export default Spacer;
