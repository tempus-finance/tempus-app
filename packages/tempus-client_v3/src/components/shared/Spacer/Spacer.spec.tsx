import { render } from '@testing-library/react';
import Spacer, { SpacerProps } from './Spacer';

const defaultProps: SpacerProps = {
  size: 16,
  variant: 'horizontal',
};

const subject = (props: SpacerProps) => render(<Spacer {...props} />);

describe('Spacer', () => {
  (['horizontal', 'vertical', 'box'] as ('horizontal' | 'vertical' | 'box')[]).forEach(variant => {
    it(`render a 16px spacer with variant ${variant}`, () => {
      const props = { ...defaultProps, variant, size: 16 };
      const { container } = subject(props);

      expect(container).not.toBeNull();
      expect(container).toMatchSnapshot();
    });

    it(`render a 32px spacer with variant ${variant}`, () => {
      const props = { ...defaultProps, variant, size: 32 };
      const { container } = subject(props);

      expect(container).not.toBeNull();
      expect(container).toMatchSnapshot();
    });
  });
});
