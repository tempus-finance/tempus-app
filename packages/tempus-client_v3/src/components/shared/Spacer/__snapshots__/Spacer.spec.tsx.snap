// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spacer render a 16px spacer with variant box 1`] = `
<div>
  <div
    style="width: 16px; height: 16px;"
  />
</div>
`;

exports[`Spacer render a 16px spacer with variant horizontal 1`] = `
<div>
  <div
    style="width: 16px; height: 0px;"
  />
</div>
`;

exports[`Spacer render a 16px spacer with variant vertical 1`] = `
<div>
  <div
    style="width: 0px; height: 16px;"
  />
</div>
`;

exports[`Spacer render a 32px spacer with variant box 1`] = `
<div>
  <div
    style="width: 32px; height: 32px;"
  />
</div>
`;

exports[`Spacer render a 32px spacer with variant horizontal 1`] = `
<div>
  <div
    style="width: 32px; height: 0px;"
  />
</div>
`;

exports[`Spacer render a 32px spacer with variant vertical 1`] = `
<div>
  <div
    style="width: 0px; height: 32px;"
  />
</div>
`;
